File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectPand.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;

/**
 * ZaakobjectPand
 */
public class ZaakobjectPand extends ZaakobjectMetObjectIdentificatie<ObjectPand> {

    /**
     * Constructor for JSONB deserialization
     */
    public ZaakobjectPand() {
    }

    /**
     * Constructor with all required fields.
     */
    public ZaakobjectPand(final URI zaak, final URI bagobjectUri, final ObjectPand pand) {
        super(zaak, bagobjectUri, ObjectTypeEnum.PAND, pand);
    }

    @Override
    public String getWaarde() {
        return getObjectIdentificatie().getIdentificatie();
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectPand.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI
import java.util.UUID

class ZaakobjectPand : ZaakobjectMetObjectIdentificatie<ObjectPand> {
    /**
     * Constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with all required fields.
     */
    constructor(zaak: URI, bagobjectUri: URI?, pand: ObjectPand?, url: URI, uuid: UUID) :
        super(zaak, bagobjectUri, ObjectTypeEnum.PAND, pand, url, uuid)

    override val waarde: String?
        get() = objectIdentificatie?.identificatie
}
